File: backend/app/models/market_price.py
# app/models/market_price.py
from sqlalchemy import Column, String, Float, Date
from sqlalchemy.dialects.postgresql import UUID
from app.db.database import Base
import uuid

class MarketPrice(Base):
    __tablename__ = "market_prices"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    commodity = Column(String, nullable=False)
    state = Column(String, nullable=False)
    mandi = Column(String, nullable=False)

    min_price = Column(Float, nullable=False)
    max_price = Column(Float, nullable=False)
    modal_price = Column(Float, nullable=False)

    arrival_date = Column(Date, nullable=False)

File: mobile_app/lib/features/market/models/market_price.dart
class MarketPrice {
  final String commodity;
  final String mandi;
  final double minPrice;
  final double maxPrice;
  final double modalPrice;
  final String date;

  MarketPrice({
    required this.commodity,
    required this.mandi,
    required this.minPrice,
    required this.maxPrice,
    required this.modalPrice,
    required this.date,
  });

  factory MarketPrice.fromJson(Map<String, dynamic> json) {
    return MarketPrice(
      commodity: json['commodity'],
      mandi: json['mandi'],
      minPrice: (json['min_price'] as num).toDouble(),
      maxPrice: (json['max_price'] as num).toDouble(),
      modalPrice: (json['modal_price'] as num).toDouble(),
      date: json['arrival_date'],
    );
  }
}
